frontend/src/components/events.jsx

```diff
@@ -10,116 +11,103 @@ class Events extends Component {
     }
 
     state = {
-        text: "",
         updateEventId: null,
         title: "",
-        description: ""
+        description: "",
+        start_date: "",
+        start_time: "",
+        end_date: "",
+        end_time: "",
+        event_image: null
+
     }
 
     resetForm = () => {
-        this.setState({ text: "", updateEventId: null });
+        this.setState({ title: "", description: "", start_date: "", start_time: "", end_date: "", end_time: "", event_image: "", updateEventId: null });
+        // this.props.events = [];
+        // this.props.fetchEvents();
+
     }
 
     submitEvent = (e) => {
         e.preventDefault();
         if (this.state.updateEventId === null) {
-            this.props.addEvent(this.state.title).then(this.resetForm)
+            this.props.addEvent(
+                this.state.title,
+                this.state.description,
+                this.state.start_date,
+                this.state.start_time,
+                this.state.end_date,
+                this.state.end_time
+            ).then(
+                this.resetForm
+            )
         } else {
-            this.props.updateEvent(this.state.updateEventId, this.state.text).then(this.resetForm);
+            this.props.updateEvent(
+                this.state.updateEventId,
+                this.state.title,
+                this.state.description,
+                this.state.start_date,
+                this.state.start_time,
+                this.state.end_date,
+                this.state.end_time
+            ).then(
+                this.resetForm
+            );
         }
-        alert("You are trying to submit an event");
+    }
+
+    selectForEdit = (id) => {
```

Comment:
Similarly to my above note, moving these state keys to a nested event object would be cleaner. You could just do `this.setState({event: event});` here.